keras/src/layers/attention/multi_head_attention.py

```diff
@@ -415,9 +429,29 @@ def _compute_attention(
           attention_output: Multi-headed outputs of attention computation.
           attention_scores: Multi-headed attention weights.
         """
-        # Note: Applying scalar multiply at the smaller end of einsum improves
-        # XLA performance, but may introduce slight numeric differences in
-        # the Transformer attention head.
+        if self._flash_attention and return_attention_scores:
+            raise ValueError(
+                "Returning attention scores is not supported when flash "
+                "attention is enabled. Please disable flash attention to access"
+                " attention scores."
+            )
+        if self._flash_attention:
+            # Directly compute the attention output using flash attention
+            attention_output = ops.dot_product_attention(
+                query=query,
+                key=key,
+                value=value,
+                mask=attention_mask,
+                scale=self._inverse_sqrt_key_dim,
+                is_causal=use_causal_mask,
+                flash_attention=self._flash_attention,
+            )
+            # Return only the attention output, as scores are not separately
+            # available
+            return attention_output, None
+
+        # Default behavior without flash attention, with explicit attention
+        # scores
```

Comment:
The 5d input tests fail if we do let  the code execute `ops.dot_product_attention` with `flash_attention=False` because it only accepts 4d inputs.  
Should we also just allow 4d inputs for MHA?

Reply:
Yes, so basically:

use_dpa_op = no dropout, no scores, ndim ==4
if use_dpa_op:
   return dot_product_attention(...)
else:
...